lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_generate_po_file_from_metadata.rb

```diff
@@ -0,0 +1,136 @@
+require_relative '../../helper/po_extended'
+require 'fastlane/action'
+require 'gettext/po'
+
+module Fastlane
+  module Actions
+    class AndroidGeneratePoFileFromMetadataAction < Action
+
+      # SPECIAL_KEYS are keys that need to be treated specially
+      SPECIAL_KEYS = %w[release_notes release_notes_previous].freeze
+      REQUIRED_KEYS_TO_COMMENT_HASH = {
+        full_description: 'Multi-paragraph text used to display in the Play Store. Limit to 4000 characters including spaces and commas!',
+        title: 'Title to be displayed in the Play Store. Limit to 30 characters including spaces and commas!',
+        short_description: 'Short description of the app to be displayed in the Play Store. Limit to 80 characters including spaces and commas!',
+        release_notes_short: nil,
+        release_notes: 'Release notes for this version to be displayed in the Play Store. Limit to 500 characters including spaces and commas!',
+        release_notes_previous: nil
+      }.freeze
+
+      REQUIRED_KEYS = REQUIRED_KEYS_TO_COMMENT_HASH.keys.map(&:to_s).freeze
+
+      OPTIONAL_KEYS_TO_LABELS_HASH = {
+        # optional_key_label: 'Description for the first app store image',
+      }.freeze
```

Comment:
> For consistency, I'd name this `OPTIONAL_KEYS_TO_COMMENT_HASH` instead of `OPTIONAL_KEYS_TO_LABELS_HASH`.

✅ 

>
> 💡 Also, the main use case for "optional" keys will mostly be screenshots. Given that, for those, unlike for the required keys that will end up being used by fastlane `deliver` (aka `upload_to_app_store`) and `supply` (aka `upload_to_play_store`) and are fixed, those optional keys will likely depend on the needs each client app.
> 
> As such:
> 
> * Unlike for the `REQUIRED_KEYS_TO_COMMENT_HASH` which will be the same for all apps, those should probably be provided by the call site instead of being hardcoded in release-toolkit, since those optional keys will come from `other_sources` and not be really standardized
> * Though as a future improvement, I'd also suggest to make those `*_KEYS_TO_COMMENT_HASH` constants use Regular Expressions matching instead of hardcoded string, so that we can imagine providing a mapping with something probably like `/^screenshots-(.*)/ => 'Text that will be used to decorate the screenshot \1'` that could work for all `screeenshots-*.txt` files found in `other_sources`  by default.


I was starting to think about this and I immediately hit a snag. Say I have a hash whose keys can be `strings` and `regexps`:

```
hash = {
  'foo': 'bar',
  '/(^.*)(:)(.*)/': 'whatever'
}
```
When looping through it how can I tell whether I am messing with a `string` or a `regexp` in ruby?

Reply:
> Your suggestion is beautiful and I really like allowing more flexibility through `regexps`. As result the code should get much cleaner.
> 
> > When trying to determine which translator comment to add for a given key, loop over all the key-value pairs of the keys_to_comments_map to .find the first one that matches one of the RegExp provided as keys. PS: We can consider, for ease of use at call site, to allow keys to be either RegExp or String, and if it's String we'll compare with the current key using a simple ==, but if it's a RegExp, we'll compare using match?`, to allow the more common case of fixed file names to be easier to write.
> 
> I love this.

@AliSoftware @jkmassel @mokagio 
I have started implementing this on this **Part3** [PR](https://github.com/Riziero/release-toolkit/pull/10). I do not expect to finish it all up however there's good progress already!

Reply:
> * Make the `other_sources` `ConfigItem` not be just an array of directories, but instead a Hash of `RexExp => comment` mappings. That will allow maximum flexibility, as those `RegExps` could either define generic patterns, e.g. `{ %r(fastlane/metadata/android/default/screenshots-(.*)\.txt) => 'Copy used to decorate screenshot \1', … }` which could either point at a list of individual files that match the pattern (like above), or whole directories (in which case all the files in that directory would have the same translator comment)…

This is a good idea however I am not sure how feasible it is. Once you have `regexp = %r{fastlane/metadata/android/default/screenshots-(.*)\.txt}` how do you use it? Get all files under `fastlane` (it could be **many**) and see if they match the `regexp`? Or perhaps it's somehow possible to extract the folder `fastlane/metadata/android/default` from the `regex` so that we can do 
```
Dir[File.join('fastlane', '*.*')].each do |file|
  if file.match? regexp
end
``` 
How about we find a compromise that makes it easier to write the code later? Something like:
```
run_described_fastlane_action(
  metadata_directory: dir,
  release_version: '1.0',
  other_sources: {
    ['fastlane/metadata/android/default/', /screenshot_(.*)/i] => 'Comment coming from `other_sources`. Text that will be used to decorate the screenshot %<shot>s'
  }
)
```
So I am suggesting to pass in a hash whose keys are [path, Regexp].